File: app/src/main/java/com/scientists/happy/botanist/ui/ShopActivity.java
// Shop for botanist accessories
// @author: Wendy Zhang and Iskander Gaba
package com.scientists.happy.botanist.ui;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import com.scientists.happy.botanist.R;

public class ShopActivity extends AppCompatActivity {
    private static final String SHOP_FERTILIZER_URL = "https://www.amazon.com/Best-Sellers-Patio-Lawn-Garden-Fertilizers/zgbs/lawn-garden/3752891";
    private static final String SHOP_ACCESSORIES_URL = "https://www.amazon.com/Best-Sellers-Patio-Lawn-Garden-Plant-Container-Accessories/zgbs/lawn-garden/3480695011";
    private static final String SHOP_POT_URL = "https://www.amazon.com/Best-Sellers-Patio-Lawn-Garden-Gardening-Pots-Planters-Accessories/zgbs/lawn-garden/3480694011";
    /**
     * Launch the activity
     * @param savedInstanceState - current app state
     */
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_shop);
        View fertilizeButton = findViewById(R.id.shop_fertilizer_button);
        fertilizeButton.setOnClickListener(new View.OnClickListener() {
            /**
             * User clicked buy now
             * @param v - current view
             */
            @Override
            public void onClick(View v) {
                openWebPage(SHOP_FERTILIZER_URL);
            }
        });
        View accessoryButton = findViewById(R.id.shop_accessories_button);
        accessoryButton.setOnClickListener(new View.OnClickListener() {
            /**
             * User clicked buy now
             * @param v - current view
             */
            @Override
            public void onClick(View v) {
                openWebPage(SHOP_ACCESSORIES_URL);
            }
        });
        View potButton = findViewById(R.id.shop_pot_button);
        potButton.setOnClickListener(new View.OnClickListener() {
            /**
             * User clicked buy now
             * @param v - current view
             */
            @Override
            public void onClick(View v) {
                openWebPage(SHOP_POT_URL);
            }
        });
    }

    /**
     * User pressed the back button
     * @return Returns true
     */
    @Override
    public boolean onSupportNavigateUp() {
        super.onBackPressed();
        return true;
    }

    /**
     * Launch Web browser
     * @param url - the url to view
     */
    private void openWebPage(String url) {
        Intent viewIntent = new Intent("android.intent.action.VIEW", Uri.parse(url));
        startActivity(viewIntent);
    }
}

File: app/src/main/java/com/scientists/happy/botanist/ui/ShopActivity.kt
// Shop for botanist pots, tools, accessories and gifts
// @author: Wendy Zhang and Iskander Gaba
package com.scientists.happy.botanist.ui

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Button
import com.scientists.happy.botanist.R

class ShopActivity : AppCompatActivity() {
    /**
     * Launch the activity
     * @param savedInstanceState - current app state
     */
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_shop)
        val potsButton = findViewById<Button>(R.id.shop_pots_button)
        potsButton.setOnClickListener{ openWebPage(SHOP_POTS_URL) }
        val toolsButton = findViewById<Button>(R.id.shop_tools_button)
        toolsButton.setOnClickListener{ openWebPage(SHOP_TOOLS_URL) }
        val giftsButton = findViewById<Button>(R.id.shop_gifts_button)
        giftsButton.setOnClickListener{ openWebPage(SHOP_GIFTS_URL) }
    }

    /**
     * User pressed the back button
     * @return Returns true
     */
    override fun onSupportNavigateUp(): Boolean {
        super.onBackPressed()
        return true
    }

    /**
     * Launch Web browser
     * @param url - the url to view
     */
    private fun openWebPage(url: String) {
        val viewIntent = Intent("android.intent.action.VIEW", Uri.parse(url))
        startActivity(viewIntent)
    }

    companion object {
        private val SHOP_POTS_URL = "http://www.crocus.co.uk/pots/"
        private val SHOP_TOOLS_URL = "http://www.crocus.co.uk/tools/"
        private val SHOP_GIFTS_URL = "http://www.crocus.co.uk/gifts-and-home/"
    }
}